papermill/preprocess.py

```diff
@@ -0,0 +1,180 @@
+from __future__ import unicode_literals, print_function
+
+import datetime
+
+from concurrent import futures
+from nbconvert.preprocessors import ExecutePreprocessor
+from nbconvert.preprocessors.execute import CellExecutionError
+
+from .iorw import write_ipynb
+
+# tqdm creates 2 globals lock which raise OSException if the execution
+# environment does not have shared memory for processes, e.g. AWS Lambda
+try:
+    from tqdm import tqdm
+    no_tqdm = False
+except OSError:
+    no_tqdm = True
+
+PENDING = "pending"
+RUNNING = "running"
+COMPLETED = "completed"
+
+
+class PapermillExecutePreprocessor(ExecutePreprocessor):
+    """Module containing a preprocessor that executes the code cells
+    and updates outputs"""
+
+    # Copyright (c) IPython Development Team.
+    # Distributed under the terms of the Modified BSD License.
+
+    # TODO: Delete this wrapper when nbconvert allows for setting preprocessor
+    # hood in a more convienent manner
+    def preprocess(self, nb, resources):
+        """
+        Copied with one edit of super -> papermill_preprocessor from nbconvert
+
+        Preprocess notebook executing each code cell.
+
+        The input argument `nb` is modified in-place.
+
+        Parameters
+        ----------
+        nb : NotebookNode
+            Notebook being executed.
+        resources : dictionary
+            Additional resources used in the conversion process. For example,
+            passing ``{'metadata': {'path': run_path}}`` sets the
+            execution path to ``run_path``.
+
+        Returns
+        -------
+        nb : NotebookNode
+            The executed notebook.
+        resources : dictionary
+            Additional resources used in the conversion process.
+        """
+        path = resources.get('metadata', {}).get('path', '')
+        if path == '':
```

Comment:
for more pythonicness: `if not path`.

Reply:
or if you want it to be strict, `if path is ''` since strings are immutable.

Reply:
PEP8 would like you to make use of the falsey-ness of empty strings https://www.python.org/dev/peps/pep-0008/#programming-recommendations not that I am a orthodox PEP8 follower or anything *coughcough* ;)

Reply:
We should probably pick one path, so I'll line up with the PEP8 recommendation of `if not path`.

Reply:
I'm just removing the if statement `path = resources.get('metadata', {}).get('path') or None`